high-level-caiman/src/lower/sched_hir/analysis/tags.rs

```diff
@@ -139,18 +139,48 @@ impl TagInfo {
 /// Tag analysis for determining tags
 /// Top: empty set
 /// Meet: union
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone)]
 #[allow(clippy::module_name_repetitions)]
-pub struct TagAnalysis {
+pub struct TagAnalysis<'a> {
     tags: HashMap<String, TagInfo>,
     specs: Rc<Specs>,
     /// For an output fact, thse are the input tags to be overridden
     input_overrides: HashMap<String, Vec<Tag>>,
+    ctx: &'a Context,
+}
+
+impl<'a> std::fmt::Debug for TagAnalysis<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut tags = self.tags.iter().collect::<Vec<_>>();
+        tags.sort_by_key(|(k, _)| (*k).clone());
```

Comment:
This line makes me feel icky...

Reply:
Agreed, the dereference and clone is a bit weird. I was trying to deal with some lifetime weirdness but I think a better method would be to just take the keys (without the values) and sort them. Or were you opposed to the whole sorting business in general? This is just done for deterministic tests. I suppose I could just use a `BTreeMap` if you'd rather that.